### DISCUSSION TITLE
Doubt about BeamDyn.

### DISCUSSION THREAD

**Discussion by aitorplaza** [OpenFAST/openfast#1435] [Q&A] [answered]
Hello to all,

I've a doubt about two outputs of BeamDyn. Is this velocity `TVxg` related to this node position `AbsXg`? is it its time derivative with respect to the ground? 

what is the difference **between** node and **sectional**?

Thanks in advance

**Comment by jjonkman** [comment, marked as answer]
Dear @aitorplaza,

It sounds like you are referring to BeamDyn's nodal outputs, in which case I agree that `TVxg` is the time-derivative of `AbsXg`.  In BeamDyn, nodes are located at a given blade cross section, so the terms "node" and "section" in the BeamDyn documentation can be viewed as interchangeable. 

Best regards,

**Reply by aitorplaza** [reply]
Dear @jjonkman.

 Thanks for your response, as always it is very clear.

**Comment by mohsamaz** [comment]
Dear @jjonkman,
first of all thank you for all of your help!
I have a blade model where the available information is given as follows:

Chord [m],Twist [deg],Thickness [%],Neutral axis x [m],Neutral axis y [m],Neutral axis local x [%],Neutral axis local y [%],User axis x [%],User axis y [%],Centre of mass x [%],Centre of mass y [%],Mass axis orientation [deg],Mass/unit length [kg/m],Polar inertia/unit length [kgm],Radii of gyration ratio [-],Principal axis orientation [deg],Shear centre x [%],Shear centre y [%],Bending stiffness about xp [Nm²],Bending stiffness about yp [Nm²],Torsional stiffness [Nm²],Axial stiffness [N]

My questions are:
1-why are the edgewise and flapwise stiffness values usually not given explicitly in blade data? can assume?
2-From these parameters, can one calculate the full 6×6 sectional stiffness and mass matrices required by BeamDyn  for each blade element?
3-Is there any existing code or tool available that can generate these matrices directly(without knowing more information)?

Any references, workflows, or example implementations would be greatly appreciated.
Best regards,

**Reply by jjonkman** [reply]
Dear @mohsamaz,

Regarding (1), you haven't said what the xp and yp axes are, but I assume the edgewise and flapwise stiffness values are equivalent to, or can be derived from, the bending stiffness about xp and yp that you do have.

Regarding (2), it sounds like you have the complete set of properties for an isotropic beam, including bending-axial coupling and shear-torsion coupling.  You can use these to generate the 6x6 matrix of the cross section, where the anisotropic terms are zero.  The document _BeamDynInput_05_2020.pdf_ attached in my post dated Jul 20, 2020 in the following topic on our forum explains: [Simulations using BeamDyn](https://forums.nrel.gov/t/simulations-using-beamdyn/2294).

If you have the composite lay-up of a cross section and want to know the 6x6 cross-section mass and stiffness matrices, cross sectional analysis tools such as PreComp, NuMAD, VABS , BECAS, ANBA, and/or SONATA could be used.

Best regards,

**Comment by mohsamaz** [comment]
Dear @jjonkman,
thank you for your reply and sharing document.

for the first question I meant why are the edgewise and flapwise shear stiffness(KshrFlp, KshrEdg) values usually not given explicitly in blade data? can assume?

Best regards,

**Reply by jjonkman** [reply]
Dear @mohsamaz,

Ah, I missed that you don't have the shear stiffness values, which is characteristic of an Euler-Bernoulli beam, as opposed to a Timoshenko beam that includes shear stiffness.

You can't derive `KShrFlp` and `KShrEdg` from the beam information you do have; rather, you'd require additional information about the cross section (such as its shape and material properties) to derive these terms.  If you don't have that, you'll likely have to estimate the shear stiffness values, e.g,. by scaling the torsion stiffness (which has the same units).

Best regards

**Comment by mohsamaz** [comment]
Dear @jjonkman,

I have a few questions regarding BeamDyn:

1) Are kp_xr, kp_yr, and kp_zr the same as the coordinates (xo, yo, zo) of the cross-section origin and in which coordinate system? is this  kp_xr, kp_yr, and kp_zr neutral axis of blade?

<img width="1386" height="315" alt="image" src="https://github.com/user-attachments/assets/5701528c-f9fb-481b-ae30-1ac226c635db" />

For BeamDyn.blade, is it correct to compute both the mass and stiffness matrices about point O (xo, yo, zo)? Should these properties always be calculated about the same point as (kp_xr, kp_yr, kp_zr), or is it acceptable to choose a different reference point?

As an example, consider the first section of the blade at the root, which is cylindrical. If O is located as shown in the figure below:
<img width="402" height="409" alt="image" src="https://github.com/user-attachments/assets/0f615525-545e-474b-b6e5-270c1e61e336" />

Here, XC is small and YC is relatively large (about 50% of the chord). This leads to the following in the stiffness matrix:

K44 = Hxx + EA·YC²

K55 = Hyy + EA·XC²

Because XC and YC differ significantly, K44 and K55 also differ, even though for a cylinder I would expect them to be nearly equal. In my blade data, the bending stiffness about xp is almost equal to that about yp, meaning Hxx ≈ Hyy.

Could you clarify why this discrepancy arises?

Best regards,

**Reply by jjonkman** [reply]
Dear @mohsamaz,

The key points in BeamDyn are used to define a spline that is used as the curvilinear reference axis, which is called point O in the _BeamDynInput_05_2020.pdf_ document.  If you follow the BeamDyn modeling guidance, the key points should be defined by a smooth curve in which case the spline will pass through each key point (but you may not get expected results if your key points do not form a smooth curve).  The curvilinear reference axis passes through a cross section at the point O of the cross sectional and the cross-sectional mass and stiffness matrices should be defined relative to this point.  This point need not be the neutral axis/tension center, shear center, or mass center.  (If the point is the neutral axis/tension center, shear center, or mass center, this means that some of the coupling terms in the mass and/or stiffness matrix will be zero.)

In your example, where point O is offset from the neutral axis/tension center , I would expect K44 to be higher than K55.  That said, I would generally recommend to define a curvilinear reference axis to be more closely aligned with the natural centers of the cross section (neutral/axis/tension center, shear center, and/or mass center) and generally follow the natural curve of the blade.  At the cylindrical blade root, I would naturally pick the centerline of this cylinder, i.e., the pitch axis.

Best regards,
